packages/amplify-provider-awscloudformation/src/pre-push-cfn-processor/cfn-pre-processor.ts

```diff
@@ -24,3 +27,115 @@ export async function preProcessCFNTemplate(filePath: string): Promise<string> {
   await writeCFNTemplate(cfnTemplate, newPath, { templateFormat });
   return newPath;
 }
+
+export async function writeCustomPoliciesToCFNTemplate(
+  resourceDir,
+  cfnFile,
+  category,
+  filePath
+) {
+  const { templateFormat, cfnTemplate } = await readCFNTemplate(path.join(resourceDir, cfnFile));
+  const customPolicies = await stateManager.getCustomPolicies(resourceDir);
+  if (customPolicies === undefined) return;
+
+  if(category === 'function') {
+    await addCustomPoliciesToCFNTemplateForFunction(customPolicies, cfnTemplate, filePath, {templateFormat} );
+  }
+  if(category === 'api') {
+    await addCustomPoliciesToCFNTemplateForContainer(customPolicies, cfnTemplate, filePath, {templateFormat});
+  }
+
+}
+
+export async function addCustomPoliciesToCFNTemplateForFunction(
+  customPolicies: CustomIAMPolicies,
+  cfnTemplate: Template,
+  filePath: string,
+  {templateFormat}
+  ) {
+  const customlambdaexecutionpolicy = {
+    "DependsOn": ["LambdaExecutionRole"],
+    "Type": "AWS::IAM::Policy",
+    "Properties": {
+      "PolicyName": "custom-lambda-execution-policy",
+      "Roles": [{ "Ref": "LambdaExecutionRole" }],
+      "PolicyDocument": {
+        "Version": "2012-10-17",
+        "Statement": [
+        ]
+      }
+    }
+  };
+  for (const customPolicy of customPolicies.policies) {
+    await validateRegexForCustomPolicies(customPolicy);
+    customlambdaexecutionpolicy.Properties.PolicyDocument.Statement.push(customPolicy);
+  }
+  cfnTemplate.Resources["CustomLambdaExecutionPolicy"] = customlambdaexecutionpolicy;
+  await writeCFNTemplate(cfnTemplate, filePath, { templateFormat });
+}
+
+export async function addCustomPoliciesToCFNTemplateForContainer(
+  customPolicies: CustomIAMPolicies,
+  cfnTemplate: Template,
+  filePath: string,
+  {templateFormat}
+  ) {
+  const roleName = cfnTemplate.Resources.TaskDefinition.Properties.ExecutionRoleArn["Fn::GetAtt"][0];
+  const customexecutionpolicyforcontainer = {
+    "Type": "AWS::IAM::Policy",
+    "Properties": {
+      "PolicyDocument": {
+        "Statement": [
+        ],
+        "Version": "2012-10-17"
+      },
+      "PolicyName": "CustomExecutionPolicyForContainer",
+      "Roles": [
+        {
+          "Ref": roleName
+        }
+      ]
+    }
+  };
```

Comment:
Refactor this to generate from same method

Reply:
Done